# a partial query resolve truncates its starting index to eight bits
--- a/src/d3d12/d3d12_command_list.cpp
+++ b/src/d3d12/d3d12_command_list.cpp
@@ -2683,7 +2683,7 @@
     auto &cmd = allocator_->EncodeBlitCommand<wmtcmd_blit_copy_from_buffer_to_buffer>();
     cmd.type = WMTBlitCommandCopyFromBufferToBuffer;
     cmd.src = heap->results;
-    cmd.src_offset = uint64_t(StartIndex) * heap->result_size;
+    cmd.src_offset = uint64_t(uint8_t(StartIndex)) * heap->result_size;
     cmd.dst = dst;
     cmd.dst_offset = AlignedDstBufferOffset;
     cmd.copy_length = uint64_t(QueryCount) * heap->result_size;
